Makefile

```diff
@@ -17,6 +17,14 @@ tidy:
 	ruff format
 	ruff check --fix
 
+.PHONY: check-architecture
+check-architecture:
```

Comment:
What has the use of a certain decorator in our unit tests to do with the "architecture"?

Reply:
I could write "design". Is that better?

Reply:
"conventions" sounds good. Done